build-logic/documentation/src/main/groovy/gradlebuild/docs/GradleJavadocsPlugin.java

```diff
@@ -145,4 +149,62 @@ private void generateJavadocs(Project project, ProjectLayout layout, TaskContain
             task.getReports().getXml().getOutputLocation().set(new File(checkstyle.getReportsDir(), "checkstyle-api.xml"));
         });
     }
+
+    /**
+     * Used to bridge Gradle 8 and Gradle 9 APIs for Gradleception.
+     *
+     * TODO: Remove after Gradle 9
+     */
+    private static class JavadocSupport {
+
+        private final Javadoc javadoc;
+
+        public JavadocSupport(Javadoc javadoc) {
+            this.javadoc = javadoc;
+        }
+
+        public void setTitle(String title) {
+            setProperty("setTitle", title);
+        }
+
+        public void setClasspath(FileCollection classpath) {
+            setProperty("setClasspath", classpath);
+        }
+
+        public void setDestinationDir(File destinationDir) {
+            setProperty("setDestinationDir", destinationDir);
+        }
+
+        private <T> void setProperty(String setterName, T value) {
+            if (GradleVersion.current().compareTo(GradleVersion.version("9.0")) < 0) {
+                setPropertyForGradleLth9(setterName, value);
+            } else {
+                setPropertyForGradleGth9(setterName, value);
+            }
+        }
+
+        private <T> void setPropertyForGradleGth9(String setterName, T value) {
```

Comment:
🤔 We could use some more readable names for these methods, like `setPropertyPostGradle9()` vs. `setPropertyPreGradle9()`. WDYT?

Reply:
Sure, I can rename them

Reply:
Done